app/helpers/discussions_helper.rb

```diff
@@ -0,0 +1,149 @@
+module DiscussionsHelper
+  def discussions_link(item)
+    link_to t(:solve_your_doubts), polymorphic_path([item, :discussions], default_discussions_params)
+  end
+
+  def solve_discussions_link
+    link_to fixed_fa_icon('comments', text: t(:solve_doubts)), discussions_path(solve_discussion_params_for(current_user))
+  end
+
+  def solve_discussion_params_for(user)
+    if user&.moderator?
+      {status: :pending_review, sort: :created_at_asc}
+    else
+      {status: :opened, sort: :created_at_asc}
+    end
+  end
+
+  def default_discussions_params
+    {status: :solved, sort: :upvotes_count_desc}
+  end
+
+  def user_avatar(user, image_class='')
+    image_tag user.image_url, height: 40, class: "img-circle #{image_class}"
+  end
+
+  def discussions_link_with_teaser(item)
+    %Q{
+      <div>
+        <h3>#{t(:discussions)}</h3>
+        <p>
+          #{t(:solve_your_doubts_teaser)}
+          #{discussions_link(item)}
+        </p>
+      </div>
+    }.html_safe
+  end
+
+  def discussion_messages_icon(discussion)
+    %Q{
+      <span class="discussion-icon fa-stack fa-xs">
+        <i class="fa fa-comment-o fa-stack-2x"></i>
+        <i class="fa fa-stack-1x">#{discussion.messages.size}</i>
+      </span>
+    }.html_safe
+  end
+
+  def discussion_upvotes_icon(discussion)
+    if discussion.upvotes_count > 0
+      %Q{
+        <span class="discussion-icon fa-stack fa-xs">
+          <i class="fa fa-star-o fa-stack-2x"></i>
+          <i class="fa fa-stack-1x">#{discussion.upvotes_count}</i>
+        </span>
+      }.html_safe
+    end
+  end
+
+  def discussion_update_status_button(status)
+    button_to t("to_#{status}"), polymorphic_path([@debatable, @discussion], {status: status}),
+              class: "btn btn-discussion-#{status}", :method => :put
+  end
+
+  def new_discussion_link(teaser_text, link_text)
+    %Q{
+      <h4>
+        <span>#{t(teaser_text)}</span>
+        <a>
+          <span id="discussion-create">
+            #{t(link_text)}
+          </span>
+        </a>
+      </h4>
+    }.html_safe
+  end
+
+  def discussion_count_for_status(status, discussions)
+    discussions.scoped_query_by(@filter_params, :status).by_status(status).count
+  end
+
+  def discussions_reset_query_link
+    link_to fa_icon(:times, text: t(:reset_query)), {} ,class: 'discussions-reset-query' unless @filter_params.blank?
+  end
+
+  def discussions_statuses
+    Mumuki::Laboratory::Status::Discussion::STATUSES
+  end
+
+  def discussions_languages(discussions)
+    discussions.map { |it| it.language.name }.uniq
+  end
+
+  def discussion_status_filter_link(status, discussions)
+    discussions_count = discussion_count_for_status(status, discussions)
+    if status.should_be_shown?(discussions_count)
+      link_to @filter_params.merge({status: status}), class: "#{'selected' if discussion_filter_selected?(:status, status)}" do
+         discussion_status_filter(status, discussions_count)
```

Comment:
Is there something we could delegate here? This code is complex and hard to understand